File: src/python/_1657_Determine_if_Two_Strings_Are_Close.py
class Solution:
    def closeStrings(self, word1: str, word2: str) -> bool:
        if len(word1) != len(word2):
            return False

        seen1 = {c: word1.count(c) for c in set(word1)}
        seen2 = {c: word2.count(c) for c in set(word2)}

        if seen1.keys() != seen2.keys():
            return False

        freq1 = {f: list(seen1.values()).count(f) for f in seen1.values()}
        freq2 = {f: list(seen2.values()).count(f) for f in seen2.values()}

        if freq1 != freq2:
            return False

        return True

File: src/java/_1657_Determine_if_Two_Strings_Are_Close.java
class Solution {
    public boolean closeStrings(String word1, String word2) {
        if (word1.length() != word2.length())
            return false;

        int[] seen1 = new int[26];
        int[] seen2 = new int[26];

        for (char c : word1.toCharArray())
            seen1[c - 'a']++;

        for (char c : word2.toCharArray())
            seen2[c - 'a']++;

        for (int i = 0; i < 26; i++)
            if (seen1[i] * seen2[i] == 0 && seen1[i] + seen2[i] != 0)
                return false;

        Arrays.sort(seen1);
        Arrays.sort(seen2);

        for (int i = 0; i < 26; i++)
            if (seen1[i] != seen2[i])
                return false;

        return true;
    }
}
